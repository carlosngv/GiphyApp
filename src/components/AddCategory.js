import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {

    const [inputValue, setinputValue] = useState('')

    const _handleOnChange = (e) => {
        setinputValue(e.target.value)
    }

    const _handleSubmit = (e) => {
        e.preventDefault();
        if(inputValue.trim().length > 2) {
            setCategories((c) => [ inputValue, ...c]);
            setinputValue('');
        }
    }

    return (
        <>
        <form onSubmit={_handleSubmit}>
            <input
                type="text"
                className="input is-rounded"
                value={inputValue}
                onChange={_handleOnChange}
                placeholder='Gif'
            />
        </form>
        </>
    )
}


AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}