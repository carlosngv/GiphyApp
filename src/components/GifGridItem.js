import React from 'react'
import PropTypes from 'prop-types';

export const GifGridItem = ({ title, url }) => {
    return (
        <div className="card animate__animated animate__backInLeft">
            <div className="card-image">
                <figure className="image is-4by3">
                    <img src={url} alt={title} />
                </figure>
            </div>
            <div className="card-content">
                <div className="content">
                    <p>{title}</p>
                </div>
            </div>
        </div>
    )
}


GifGridItem.propTypes =  {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}