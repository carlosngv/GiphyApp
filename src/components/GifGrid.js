import React, { Fragment} from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    const { data: images, loading } = useFetchGifs(category);
         
    return (
        <Fragment>
            <h1 className="category-title animate__animated animate__backInRight">{category}</h1>
            <div className="card-grid">
                    {loading && 'Loading...'} 
                    {
                        
                        images.map((img) => (
                            <GifGridItem
                                key={img.id}
                                {...img}
                            />
                        ))
                    }
            </div>
        </Fragment>
    )
}
