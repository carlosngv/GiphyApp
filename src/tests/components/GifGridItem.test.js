import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';
import toJson from 'enzyme-to-json';


describe('GifGridItem tests', () => {
    const title = 'Test item';
    const url = 'https://testurl.com';
    const wrapper = shallow(<GifGridItem title={title} url={url} />);
    
    test('Should show component correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should contain text within p tag', () => {
        const pTagContent = wrapper.find('p').text().trim();
        expect(pTagContent).toBe(title);
    });
    
    test('img should have same url src and same alt title', () => {
        const img = wrapper.find('img');
        const {src, alt} = img.props();
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });
    
    test('parent div should contain animate__backInLeft class', () => {
        const div = wrapper.find('div').at(0);
        const { className } = div.props();
        const classname = "animate__backInLeft";
        expect(className.includes(classname)).toBe(true);
    });
    
});

