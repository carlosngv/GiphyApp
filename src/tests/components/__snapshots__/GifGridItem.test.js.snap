// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGridItem tests Should show component correctly 1`] = `
<div
  className="card animate__animated animate__backInLeft"
>
  <div
    className="card-image"
  >
    <figure
      className="image is-4by3"
    >
      <img
        alt="Test item"
        src="https://testurl.com"
      />
    </figure>
  </div>
  <div
    className="card-content"
  >
    <div
      className="content"
    >
      <p>
        Test item
      </p>
    </div>
  </div>
</div>
`;
